src/app/js/fields/FieldComposedOf.js

```diff
@@ -0,0 +1,188 @@
+                            onChange={handleSelectColumn(index)}
+                            style={styles.select}
+                            hintText={polyglot.t('select_a_column')}
+                            value={col}
+                            required
+                        >
+                            {fields.map(f => (
+                                <MenuItem
+                                    className={`composite-field-${index}-${
+                                        f.name
+                                    }`}
+                                    key={`composite-field-${index}-${f.name}`}
+                                    value={f.name}
+                                    primaryText={f.label}
+                                />
+                            ))}
+                        </SelectField>
+                        {index > 1 && (
+                            <IconButton
+                                className={`btn-remove-composite-field btn-remove-composite-field-${
+                                    index
+                                }`}
+                                onClick={handleRemoveColumn(index)}
+                                title={polyglot.t('remove_composition_column')}
+                            >
+                                <IconDelete />
+                            </IconButton>
+                        )}
+                    </div>
+                ))}
+            {isComposedOf && (
+                <FlatButton
+                    className="btn-add-composition-column"
+                    label={polyglot.t('add_composition_column')}
+                    onClick={handleAddColumn}
+                />
+            )}
+        </div>
+    </div>
+);
+
+FieldComposedOf.propTypes = {
+    isComposedOf: PropTypes.bool,
+    columns: PropTypes.arrayOf(PropTypes.string),
+    fields: PropTypes.arrayOf(fieldPropTypes).isRequired,
+    handleAddColumn: PropTypes.func.isRequired,
+    handleSelectColumn: PropTypes.func.isRequired,
+    handleRemoveColumn: PropTypes.func.isRequired,
+    handleToggle: PropTypes.func.isRequired,
+    p: polyglotPropTypes.isRequired,
+};
+
+FieldComposedOf.defaultProps = {
+    isComposedOf: false,
+    columns: [],
+};
+
+const mapStateToProps = (state, { FORM_NAME }) => {
+    const composedOf = formValueSelector(FORM_NAME || FIELD_FORM_NAME)(
+        state,
+        'composedOf',
+    );
+
+    if (composedOf && composedOf.fields && composedOf.fields.length > 1) {
+        return {
+            isComposedOf: composedOf.isComposedOf,
+            columns: composedOf.fields,
+        };
+    }
+
+    return { isComposedOf: false, columns: [] };
+};
+
+const mapDispatchToProps = (dispatch, { FORM_NAME = FIELD_FORM_NAME }) => ({
+    onChange: composedOf => {
+        dispatch(change(FORM_NAME, 'composedOf', composedOf));
+    },
+});
+
+export default compose(
+    connect(mapStateToProps, mapDispatchToProps),
+    withHandlers({
```

Comment:
not a big fan, this component is complex enough to justify using the class syntax, and therefore avoid the usage of `withHandlers`